libraries/AP_GPS/AP_GPS.cpp

```diff
@@ -101,8 +101,8 @@ const AP_Param::GroupInfo AP_GPS::var_info[] = {
 #if GPS_MAX_RECEIVERS > 1
     // @Param: AUTO_SWITCH
     // @DisplayName: Automatic Switchover Setting
-    // @Description: Automatic switchover to GPS reporting best lock
-    // @Values: 0:Disabled,1:UseBest,2:Blend,3:UseSecond
+    // @Description: Automatic switchover to GPS reporting best lock, 1:UseBest selects the GPS with highest status, if both are equal the GPS with highest satellites is used. 4:UseFirst conservative uses the GPS with the highest status, if both are equal the first will be used until it has 4 satellites fewer, as soon first has more it will be used again
+    // @Values: 0:Disabled,1:UseBest,2:Blend,3:UseSecond,4:UseFirst conservative switch
```

Comment:
add GPS_PRIMARY, default to -1, if not -1, then use the primary GPS if it has a reasonable lock

Reply:
I have added GPS_PRIMARY defaults to 0 for first, removed the 3:UseSecond GPS_AUTO_SWITCH option and done the param conversion to set GPS_PRIMARY from that